problem-spring-common/src/main/java/org/zalando/problem/spring/common/AdviceTrait.java

```diff
@@ -0,0 +1,141 @@
+package org.zalando.problem.spring.common;
+
+import org.apiguardian.api.API;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.zalando.problem.*;
+
+import javax.annotation.Nullable;
+import java.net.URI;
+import java.util.Optional;
+
+import static java.util.Arrays.asList;
+import static org.apiguardian.api.API.Status.MAINTAINED;
+import static org.apiguardian.api.API.Status.STABLE;
+import static org.springframework.core.annotation.AnnotatedElementUtils.findMergedAnnotation;
+import static org.zalando.problem.spring.common.Lists.lengthOfTrailingPartialSubList;
+import static org.zalando.problem.spring.common.MediaTypes.PROBLEM;
+
+/**
+ * <p>
+ * Advice traits are simple interfaces that provide a single method with a default
+ * implementation. They are used to provide {@link ExceptionHandler} implementations to be used in
+ * Spring Controllers and/or in a {@link ControllerAdvice}. Clients can choose which traits they what to
+ * use à la carte.
+ * </p>
+ * <p>
+ * Advice traits are grouped in packages, based on they use cases. Every package has a composite advice trait that
+ * bundles all traits of that package.
+ * </p>
+ *
+ * @see ControllerAdvice
+ * @see ExceptionHandler
+ * @see Throwable
+ * @see Exception
+ * @see Problem
+ */
+@API(status = STABLE)
```

Comment:
I'd mark everything in `common` as `@API(status = INTERNAL)` for now.

Reply:
Those functions were before in `spring-web`. So is it not breaking to mark them now as `INTERNAL`

Reply:
This class (a) has been extracted from the former `AdviceTrait` (b), i.e. b extends a. If a is INTERNAL but b is STABLE than every method in a is effectively STABLE, as long as it's accessed via b's interface/API.

Reply:
OK 👍